packages/libs/src/sdk/scripts/manageRewardsLookupTable.ts

```diff
@@ -0,0 +1,396 @@
+      await connection.getLatestBlockhash()
+    const transaction = new VersionedTransaction(
+      new TransactionMessage({
+        payerKey: payer.publicKey,
+        recentBlockhash: blockhash,
+        instructions: [extendInstruction]
+      }).compileToV0Message()
+    )
+    transaction.sign([payer])
+    const signature = await connection.sendTransaction(transaction)
+    console.info('Confirming...')
+    await connection.confirmTransaction({
+      signature,
+      blockhash,
+      lastValidBlockHeight
+    })
+    console.info('Confirmed.')
+  }
+}
+
+/**
+ * Get the proper config for the given environment
+ */
+const getConfig = (environment: SdkConfig['environment']) => {
+  const config =
+    environment === 'development'
+      ? developmentConfig
+      : environment === 'staging'
+      ? stagingConfig
+      : productionConfig
+  return config
+}
+
+/**
+ * Creates the lookup table for the given environment with the given keypair as
+ * both payer and authority.
+ *
+ * After running, be sure to update `generateServicesConfig.ts` with the address
+ * f the lookup table.
+ */
+const createLookupTable = async ({
+  environment,
+  keypair
+}: {
+  environment: SdkConfig['environment']
+  keypair: string
+}) => {
+  console.info(
+    'Creating rewards lookup table for',
+    environment,
+    'using keypair',
+    keypair
+  )
+  const wallet = Keypair.fromSecretKey(
+    Uint8Array.from(
+      JSON.parse(
+        readFileSync(
+          path.resolve(untildify(keypair ?? '~/.config/solana/id.json')),
+          { encoding: 'utf-8' }
+        )
+      )
+    )
+  )
+  const sdk = audiusSdk({
+    appName: 'generate-rewards-lookup-table',
+    environment,
+    services: {
+      logger: new Logger()
+    }
+  })
+  const config = getConfig(environment)
+  const connection = sdk.services.solanaClient.connection
+
+  const rewardManagerState = new PublicKey(
+    config.solana.rewardManagerStateAddress
+  )
+  const rewardManagerProgramId = new PublicKey(
+    config.solana.rewardManagerProgramAddress
+  )
+  const rewardManagerAuthority = RewardManagerProgram.deriveAuthority({
+    programId: rewardManagerProgramId,
+    rewardManagerState
+  })
+  const rewardManagerTokenSource = (
+    await sdk.services.rewardManagerClient.getRewardManagerState()
+  ).tokenAccount
+
+  const addresses = [
+    SystemProgram.programId,
+    SYSVAR_RENT_PUBKEY,
+    SYSVAR_INSTRUCTIONS_PUBKEY,
+    TOKEN_PROGRAM_ID,
+    TOKEN_2022_PROGRAM_ID,
+    rewardManagerState,
+    rewardManagerAuthority,
+    rewardManagerTokenSource
+  ]
+
+  const payer = wallet.publicKey
+  const authority = wallet.publicKey
+
+  const slot = await connection.getSlot()
+  const { blockhash, lastValidBlockHeight } =
+    await connection.getLatestBlockhash()
+  const [createLookupTableInstruction, lookupTableAddress] =
+    AddressLookupTableProgram.createLookupTable({
+      authority,
+      payer,
+      recentSlot: slot - 1
```

Comment:
why the -1?

Reply:
> Note that using the most recent slot sometimes results in an error after sending the transaction. To avoid this, you can use a slot that is one slot prior the most recent one (e.g. recentSlot: slot - 1). However, if you still encounter an error when sending the transaction, you can try resending the transaction.

https://solana.com/developers/courses/program-optimization/lookup-tables#create-a-lookup-table

Reply:
In the future, maybe add a comment for things like this? It's going to be cryptic looking at this code later on. :-)